--- /mnt/data/htk34/httptoolkit-server-main/src/interceptors/electron.ts	2026-04-24 16:49:31.000000000 +0000
+++ /mnt/data/htk33/httptoolkit-server-main/src/interceptors/electron.ts	2026-04-25 12:04:56.000000000 +0000
@@ -90,7 +90,9 @@
                 // Try to close the debug connection if open, but very carefully
                 try {
                     await debugClient.close();
-                } catch (e) { }
+                } catch (e) {
+                    console.warn('Failed to cleanly close Electron debug client after process error:', e);
+                }
             }
 
             // If we're still in the process of debugging the app, give up.
@@ -223,4 +225,4 @@
         );
     }
 
-}
\ No newline at end of file
+}
