--- /mnt/data/htk34/httptoolkit-server-main/src/util/fs.ts	2026-04-24 16:49:31.000000000 +0000
+++ /mnt/data/htk33/httptoolkit-server-main/src/util/fs.ts	2026-04-25 12:04:56.000000000 +0000
@@ -1,6 +1,8 @@
+import { promisify } from 'util';
 import * as fs from 'fs';
 import * as path from 'path';
 import * as tmp from 'tmp';
+import rimraf from 'rimraf';
 import { lookpath } from 'lookpath';
 import { isErrorLike } from '@httptoolkit/util';
 
@@ -9,8 +11,7 @@
 export const readDir = fs.promises.readdir;
 export const readLink = fs.promises.readlink;
 export const deleteFile = fs.promises.unlink;
-export const deleteFolder = (folder: string) =>
-    fs.promises.rm(folder, { recursive: true, force: true });
+export const deleteFolder = promisify(rimraf);
 export const checkAccess = fs.promises.access;
 export const chmod = fs.promises.chmod;
 export const mkDir = fs.promises.mkdir;
